elifearticle/utils.py

```diff
@@ -145,10 +145,5 @@ def author_name_from_json(author_json):
 
 def text_from_affiliation_elements(department, institution, city, country):
     "format an author affiliation from details"
-    text = ""
-    for element in (department, institution, city, country):
-        if text != "":
-            text += ", "
-        if element:
-            text += element
-    return text
+    return ', '.join(element for element in [department, institution, city, country]
+                     if element is not None)
```

Comment:
If you changed this check to be `if element` it will also catch empty strings being passed in.

e.g. at present if you passed in `'1', '', '3', '4'` you would get back `1, , 3, 4`. With the change this would output `1, 3, 4`. It might not be likely that blank strings are passed in but its an easy catch when you don't want blank entries included in the output of a `join`.

Reply:
Agreed, I missed that example, good catch @seanwiseman !